### PR TITLE
ranger: rerange file and rename some method.

### PR DESCRIPTION
**Pull request by winoros**
There're 4 files in this package.

`checker.go`: the `conditionChecker` struct which check whether one simple expression can be used to calculate range.
`detacher.go`: find useful expression which can be used to calculate range from a bunch of expression.
`points.go`: convert expression to range points.
`ranger.go`: build ranges which are specific type.

### CODE REVIEW

**Comment by winoros** [comment]
PTAL @hanfei1991 @coocood @lamxTyler @zz-jason

**Comment by coocood** [comment]
The change added 43 lines, and I can't easily find which part is added code.
Can you not modify the code and rearranging the file at the same time?

**Comment by winoros** [comment]
@coocood ok i'll do that in another pr.

**Comment by winoros** [comment]
@coocood Now the added line should be only the file header.

File: plan/physical_plan_builder.go [hunk 1]
```diff
@@ -878,7 +878,7 @@ func (is *PhysicalIndexScan) addPushedDownSelection(copTask *copTask, p *DataSou
 	if len(is.filterCondition) > 0 {
 		var indexConds, tableConds []expression.Expression
 		if copTask.tablePlan != nil {
-			indexConds, tableConds = ranger.DetachIndexFilterConditions(is.filterCondition, is.Index.Columns, is.Table)
+			indexConds, tableConds = splitIndexFilterConditions(is.filterCondition, is.Index.Columns, is.Table)
```
Comment (coocood) [inline plan/physical_plan_builder.go]: The `splitIndexFilterCondition` is not the same as `ranger.DetachIndexFitlerConditions`.

**Comment by coocood** [comment]
The PR still has modifications.

File: plan/physical_plan_builder.go (hunk 1 above)
Comment (winoros) [reply plan/physical_plan_builder.go]: @coocood  fixed.

**Comment by coocood** [comment]
LGTM

**Comment by hanfei1991** [comment]
expression-> range_point can be called builder.go
range_point -> range can be called ranger.go

**Comment by hanfei1991** [comment]
expression_ranger.go can be called detacher.go ?

**Comment by hanfei1991** [comment]
and a checker.go ...

**Pull request merged**